subsquid/src/server-extension/resolvers/picassoOverviewStats.ts

```diff
@@ -36,26 +44,45 @@ export class PicassoOverviewStatsResolver
 
   @FieldResolver({ name: "totalValueLocked", defaultValue: 0 })
   async totalValueLocked(): Promise<bigint> {
+    const wsProvider = new WsProvider(chain());
+    const api = await ApiPromise.create({ provider: wsProvider });
+
     const manager = await this.tx();
 
-    // TODO: add something like WHERE source = 'All' once PR 1703 is merged
-    let lockedValue: { amount: bigint }[] = await manager
-      .getRepository(HistoricalLockedValue)
-      .query(
-        `
-        SELECT
-          amount
-        FROM historical_locked_value
-        ORDER BY timestamp DESC
-        LIMIT 1
-      `
-      );
+    const lockedValue = await manager.getRepository(CurrentLockedValue).find({
+      where: {
+        source: LockedSource.All,
+      },
+    });
+
+    const assetPrices = lockedValue.reduce<Record<string, bigint>>(
+      (acc, value) => {
+        acc[value.assetId] = 0n;
+        return acc;
+      },
+      {}
+    );
 
-    if (!lockedValue?.[0]) {
-      return Promise.resolve(0n);
+    for (const assetId of Object.keys(assetPrices)) {
+      try {
+        const oraclePrice = await api.query.oracle.prices(assetId);
+        if (oraclePrice?.price) {
+          assetPrices[assetId] = BigInt(oraclePrice?.price?.toString() || 0n);
+        }
+        // TODO: what to do if no there's no Oracle price?
```

Comment:
return "N/A"

Reply:
In the case of this resolver it's accumulating over all the locked assets, so if one locked asset is missing its price, I can only think on ignoring it, __unless__ there's some other source I can get prices from.